primeqa/ir/dense/colbert_top/colbert/modeling/colbert.py

```diff
@@ -130,7 +167,9 @@ def doc(self, input_ids, attention_mask, keep_dims=True):
         mask = torch.tensor(self.mask(input_ids, skiplist=self.skiplist), device=self.device).unsqueeze(2).float()
         D = D * mask
 
-        D = torch.nn.functional.normalize(D, p=2, dim=2).half()
+        D = torch.nn.functional.normalize(D, p=2, dim=2)
+        if self.use_gpu:
+            D = D.half()
```

Comment:
D = D.cuda()?

Reply:
I think it's on a GPU already if a GPU is used.